server: more actionable output for freeze-cluster

tbg opened the pull request:
`./cockroach freeze-cluster` will now return (up to 10) Replicas which
failed to apply the freeze.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7104)

<!-- Reviewable:end -->

tamird commented:
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[roachpb/api.proto, line 811 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCXXS2w77oy_PErmA:-KJlCXXS2w77oy_PErmB:1618789527) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/roachpb/api.proto#L811)):*
> ```protobuf
> }
> 
> // A PollFrozenRequest asks addressed Store for its frozen or thawed Replicas.
> ```

missing article

---

*[server/admin.go, line 706 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCkFnDAHJoAvCQqvl:-KJlCkFnDAHJoAvCQqvm:1010688961) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/server/admin.go#L706)):*
> ```Go
> 							// the retry loop picks it up again.
> 							message += fmt.Sprintf("%d replicas report wrong status", numMismatching)
> 							if numMismatching > 10 {
> ```

nit: `if limit := 10; numMismatching > limit { ..` then reuse below

---

*[server/admin.go, line 707 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCwgzQVqrkZtfSGoL:-KJlCwgzQVqrkZtfSGoM:1019116829) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/server/admin.go#L707)):*
> ```Go
> 							message += fmt.Sprintf("%d replicas report wrong status", numMismatching)
> 							if numMismatching > 10 {
> 								message += " [truncated]: "
> ```

is it time to make `message` a buffer?

---

*[storage/store_test.go, line 1880 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlDRMgsKeIELTOO1sX:-KJlDRMgsKeIELTOO1sY:1540496740) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/storage/store_test.go#L1880)):*
> ```Go
> 		if len(results) != 0 {
> 			t.Fatal(util.ErrorfSkipFrames(1,
> 				"expected frozen=%t, got %d thawed replicas: %+v",
> ```

thawed? shouldn't that be conditional?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104)*
<!-- Sent from Reviewable.io -->

tbg committed:
server: more actionable output for freeze-cluster

`./cockroach freeze-cluster` will now return (up to 10) Replicas which
failed to apply the freeze.

tschottdorf commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[roachpb/api.proto, line 811 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCXXS2w77oy_PErmA:-KJlXXzH2pjS2CIrANuO:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/roachpb/api.proto#L811)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> missing article
</details>
Done.

---

*[server/admin.go, line 706 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCkFnDAHJoAvCQqvl:-KJlXspR1ZzjdX-qziW5:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/server/admin.go#L706)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> nit: `if limit := 10; numMismatching > limit { ..` then reuse below
</details>
Done.

---

*[server/admin.go, line 707 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlCwgzQVqrkZtfSGoL:-KJlXybnRSUcW4rm3HkL:-1542355953) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/server/admin.go#L707)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> is it time to make `message` a buffer?
</details>
Getting there, but not quite there yet.

---

*[storage/store_test.go, line 1880 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104#-KJlDRMgsKeIELTOO1sX:-KJlYDCB3m7XXTJrrKef:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb1052bcd87677e3ae6b6d21e6433bfb0c25e3b9/storage/store_test.go#L1880)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> thawed? shouldn't that be conditional?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 6 of 6 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7104)*
<!-- Sent from Reviewable.io -->
